### PR TITLE
Implement piechart component for account index page

### LINKED ISSUE #915: Implement piechart component for account index page

### PR DESCRIPTION
**Pull request by prin-r**
Fix #915

### CODE REVIEW

File: scan/src/components/PieChart.re [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        offset +. 180.0,
+        asize <= 180.0 ? 0.0 : asize -. 180.0,
+        color,
+        asize <= 180.0,
+      )}
+    />
+  </>;
+
+[@react.component]
+let make = (~size=187, ~availableBalance=500., ~balanceAtStake=60., ~reward=20.) => {
```
Comment (sorawit) [inline scan/src/components/PieChart.re]: Let's not have default value here. If we want to mock something, we do in AccountIndexPage

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/components/PieChart.re [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        asize <= 180.0,
+      )}
+    />
+  </>;
+
+[@react.component]
+let make = (~size, ~availableBalance, ~balanceAtStake, ~reward) => {
+  let totalBalance = availableBalance +. balanceAtStake +. reward;
+  let balanceAtStakeAsize = totalBalance == 0. ? 0. : 360. *. balanceAtStake /. totalBalance;
+  let brewardAsize = totalBalance == 0. ? 0. : 360. *. reward /. totalBalance;
```
Comment (taobun) [inline scan/src/components/PieChart.re]: What is brewardA

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/components/PieChart.re (hunk 2 above)
Comment (prin-r) [reply scan/src/components/PieChart.re]: "A" is from angle Yay!

**Comment by evilpeach** [comment]
@prin-r Please fix the conflict.

File: scan/src/components/PieChart.re [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    style([
+      visibility(isHidden ? `hidden : `visible),
+      overflow(`hidden),
+      width(`percent(100.)),
+      height(`percent(100.)),
+      position(`absolute),
+      transforms([
+        `translate((`zero, `percent(-50.))),
+        `rotate(`deg(90.)),
+        `rotate(`deg(offset)),
```
Comment (evilpeach) [inline scan/src/components/PieChart.re:25]: duplicated?

Comment (prin-r) [reply scan/src/components/PieChart.re:25]: Transformation can be combine. We just rotate 90(deg) and then rotate offset(deg).

**Review by evilpeach — CHANGES_REQUESTED** [review]
remove `pie-chart.png`

File: scan/src/components/PieChart.re [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        height(`percent(100.)),
+        position(`absolute),
+        transforms([`translate((`zero, `percent(100.))), `rotate(`deg(asize))]),
+        transformOrigin(`percent(50.), `percent(0.)),
+        backgroundColor(color),
+      ]),
+    ]);
+};
+
+let renderSegment = (offset, asize, color) =>
```
Comment (evilpeach) [inline scan/src/components/PieChart.re]: Let's use `angle` instead of `asize`.

File: scan/src/components/PieChart.re [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  <>
+    <div className={Styles.segment(offset, asize <= 180.0 ? asize : 180.0, color, false)} />
```
Comment (evilpeach) [inline scan/src/components/PieChart.re]: ditto

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/PieChart.re [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        angle <= 180.0,
+      )}
+    />
+  </>;
+
+[@react.component]
+let make = (~size, ~availableBalance, ~balanceAtStake, ~reward) => {
+  let totalBalance = availableBalance +. balanceAtStake +. reward;
+  let balanceAtStakeAsize = totalBalance == 0. ? 0. : 360. *. balanceAtStake /. totalBalance;
+  let rewardAsize = totalBalance == 0. ? 0. : 360. *. reward /. totalBalance;
```
Comment (evilpeach) [inline scan/src/components/PieChart.re]: `balanceAtStakeAngle` and `rewardAngle`

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Review by tansawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**